plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/KGraphLanguageServerExtension.xtend

```diff
@@ -0,0 +1,600 @@
+        val KlighdDataManager kdm = KlighdDataManager.instance
+        return kdm.getAvailableSyntheses(currentModelClass).map [
+            val id = kdm.getSynthesisID(it)
+            val displayedName = if (id.contains(".") && !id.endsWith(".")) {
+                    id.substring(id.lastIndexOf('.') + 1);
+                } else {
+                    id
+                }
+            return new SetSynthesesActionData(id, displayedName)
+        ].toList
+    }
+    
+    override getOptions(GetOptionParam param) {
+        return doRead(param.uri) [ resource, ci |
+            synchronized (diagramState) {
+                val ViewContext viewContext = diagramState.getKGraphContext(resource.URI.toString)
+                if (viewContext === null) {
+                    // A diagram for this file is currently not opened, so no options can be shown.
+                    return null
+                }
+                val synthesisOptions = new ArrayList<ValuedSynthesisOption>
+                val recentSynthesisOptions = diagramState.recentSynthesisOptions
+                for (option : viewContext.displayedSynthesisOptions) {
+                    val currentValue = recentSynthesisOptions.get(option)
+                    synthesisOptions.add(new ValuedSynthesisOption(option, currentValue))
+                }
+                val layoutOptionUIData = calculateLayoutOptionUIData(viewContext.displayedLayoutOptions)
+                
+                return new GetOptionsResult(synthesisOptions, layoutOptionUIData,
+                    viewContext.displayedActions)
+            }
+        ]
+    }
+    
+    def calculateLayoutOptionUIData(List<Pair<IProperty<?>, List<?>>> displayedLayoutOptions) {
+        val List<LayoutOptionUIData> layoutOptionUIData = new ArrayList
+        for (pair : displayedLayoutOptions) {
+            var Object first
+            var Object second
+            if (pair.second instanceof Collection) {
+                val iterator = (pair.second as Collection<?>).iterator
+                first = if (iterator.hasNext) iterator.next else null
+                second = if (iterator.hasNext) iterator.next else null
+            } else {
+                first = null
+                second = null
+            }
+
+            val LayoutOptionData optionData = LayoutMetaDataService.instance.getOptionData(pair.first.id)
+            if (optionData.visibility !== Visibility.HIDDEN) {
+                if (first instanceof Number && second instanceof Number) {
+                    layoutOptionUIData.add(new LayoutOptionUIData(optionData, (first as Number).floatValue,
+                        (second as Number).floatValue, null))
+                } else if (pair.second === null) {
+                    layoutOptionUIData.add(new LayoutOptionUIData(optionData, null, null, null))
+                } else {
+                    layoutOptionUIData.add(new LayoutOptionUIData(optionData, null, null, pair.second))
+                }
```

Comment:
This can be handle with one case. Is there a particular reason to add this as a distinct case?